### ISSUE TITLE
Ripples on mouse clicks cropped on the right edge and bottom edge

### ISSUE THREAD

**Issue by i-give-up** [marius-sucan/KeyPress-OSD#4]
Environment: Windows 8.1, Keypress OSD v4.30.0

I checked "Ripples" under "Preferences > Mouse". Issue happens even with default Size/Thickness/Speed.

Whenever I click, the ripples seem to be cropped at the right edge and bottom edge. Below is a series of screenshots I've taken from a screen recording. Each click is a bit different but the cropping is visible.

![mouseclick_ripples](https://user-images.githubusercontent.com/14156307/38775839-dd1829ca-40be-11e8-98b2-1a1d712548f0.jpg)

Alternatively, here's [a link to the screen recording](http://sendvid.com/24nuxinc).

It's fairly noticeable for Speed values up to 15. At Speed = 16, it's slightly noticeable. At Speed = 17, you can't really see it.

Another thing that would work is to lower Thickness to say 3 or 4, but the preferences doesn't allow you to go below 5 for that.

**Comment by marius-sucan** [comment]
Hello!

Thank you very much for your report.

To help me understand better the context, please tell me your screen DPI and DPI scale.

Best regards, Marius.

**Comment by i-give-up** [comment]
Here are the numbers

- screen DPI: 102.46 (1920x1080 on 21.5-in diagonal - not sure if calculation is correct?) 
- DPI scale: 100% (assuming you mean the value in "Control Panel\Appearance and Personalization\Display"?)

**Comment by i-give-up** [comment]
Upgraded to 4.30.1, restore defaults, and checked "Ripples". The issue is still present

---

Interestingly, I tested on a different computer and this issue is not reproducible there.

- Windows 7 64-bit
- DPI 80.36 (1366x768, diagonal 19.5-in)
- DPI scaling 100%

**Comment by marius-sucan** [comment]
Hello, again!

Please test the new version. I just updated the files.... To do so, you need to update again [the version number has not changed], answer Yes to Force Update attempt.

Hope the issue is solved.

Best regards, Marius.

**Comment by i-give-up** [comment]
Unfortunately I'm still seeing the issue. I updated to latest version using the menu, restore defaults, and checked "Ripples". [Link to screen recording](http://sendvid.com/6vd69w6t). Were you able to replicate the issue on any of your machines?

**Comment by marius-sucan** [comment]
Sorry, I was unable to reproduce the issue on the four other systems I have access to.... Two in Virtual Box [XP and Win7] and... one laptop with Win10 x64, and a desktop PC with Win7 x64.

**Comment by i-give-up** [comment]
I believe I have found a fix/workaround after a lot of trial-and-error. I have submitted [a PR with the modification I made](https://github.com/marius-sucan/KeyPress-OSD/pull/6).

The fix involves using a larger dimensions for `UpdateLayeredWindow` DLL call which seems to get rid of the cropping. I'm not sure how much to increase it by but `L + 4` works even at ripple thickness. A nicer expression would be `RippleDiameter+2*MouseRippleThickness*tf` but it would increase window size significantly for larger `MouseRippleThickness` values and I'm not sure if larger window size has any downsides.

Another thing I'm unsure of is how to name the new variable. Some of the variables in the scripts use camel case starting with uppercase, some use camel case starting with lowercase, and some use snake case.

Feel free to make any modifications that you think necessary.
